In [2]:
# Data manipulation
import pandas as pd
import numpy as np

# Visualization
import matplotlib.pyplot as plt

# Machine Learning - Models
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier, plot_tree

# Machine Learning - Data Processing
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

# Machine Learning - Evaluation
from sklearn.metrics import (
    accuracy_score,
    confusion_matrix,
    classification_report,
    precision_score,
    recall_score,
    f1_score,
    ConfusionMatrixDisplay
)

# For nicer plots
plt.style.use('seaborn-v0_8-whitegrid')

# Ignore warnings for cleaner output
import warnings
warnings.filterwarnings('ignore')

print("All libraries loaded successfully!")

All libraries loaded successfully!


In [7]:
df = pd.read_csv('./salaries.csv')
df.info()
#unique_count = df['salary_currency'].value_counts()
#print(unique_count)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88584 entries, 0 to 88583
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           88584 non-null  int64 
 1   experience_level    88584 non-null  object
 2   employment_type     88584 non-null  object
 3   job_title           88584 non-null  object
 4   salary              88584 non-null  int64 
 5   salary_currency     88584 non-null  object
 6   salary_in_usd       88584 non-null  int64 
 7   employee_residence  88584 non-null  object
 8   remote_ratio        88584 non-null  int64 
 9   company_location    88584 non-null  object
 10  company_size        88584 non-null  object
dtypes: int64(4), object(7)
memory usage: 7.4+ MB


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88584 entries, 0 to 88583
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           88584 non-null  int64 
 1   experience_level    88584 non-null  object
 2   employment_type     88584 non-null  object
 3   job_title           88584 non-null  object
 4   salary              88584 non-null  int64 
 5   salary_currency     88584 non-null  object
 6   salary_in_usd       88584 non-null  int64 
 7   employee_residence  88584 non-null  object
 8   remote_ratio        88584 non-null  int64 
 9   company_location    88584 non-null  object
 10  company_size        88584 non-null  object
dtypes: int64(4), object(7)
memory usage: 7.4+ MB

286
job_title
Data Scientist                     12388
Data Engineer                      10107
Software Engineer                   9404
Data Analyst                        7866
Machine Learning Engineer           6226

Name: count, dtype: int64

In [ ]:
# If it is 'US', keep it 'US'; otherwise, change to 'Other'
df['company_location'] = np.where(df['company_location'] == 'US', 'US', 'Other')

#Reduced the job_title to top 5 and other
top_jobs = ['Data Scientist', 'Data Engineer', 'Software Engineer', 'Data Analyst', 'Machine Learning Engineer']
df.loc[~df['job_title'].isin(top_jobs), 'job_title'] = 'Other'
#filters to USD currency values and Full Time employment type
df_cleaned = df[(df['salary_currency'] == 'USD') & (df['employment_type']=='FT')]



# One-Hot Encoding
df_cleaned = pd.get_dummies(df_cleaned, columns=['experience_level', 'company_location'], dtype=int)

df_cleaned.head()

work_year employment_type     job_title  salary salary_currency  \
1       2025              FT         Other  165000             USD   
2       2025              FT         Other  109000             USD   
3       2025              FT         Other  294000             USD   
4       2025              FT         Other  137600             USD   
5       2025              FT  Data Analyst   82000             USD   

   salary_in_usd employee_residence  remote_ratio company_size  \
1         165000                 US             0            M   
2         109000                 US             0            M   
3         294000                 US             0            M   
4         137600                 US             0            M   
5          82000                 US             0            M   

   experience_level_EN  experience_level_EX  experience_level_MI  \
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   
5                    1                    0                    0   

   experience_level_SE  company_location_Other  company_location_US  
1                    1                       0                    1  
2                    1                       0                    1  
3                    1                       0                    1  
4                    1                       0                    1  
5                    0                       0                    1

<class 'pandas.core.frame.DataFrame'>
Index: 83994 entries, 1 to 88582
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           83994 non-null  int64 
 1   experience_level    83994 non-null  object
 2   employment_type     83994 non-null  object
 3   job_title           83994 non-null  object
 4   salary              83994 non-null  int64 
 5   salary_currency     83994 non-null  object
 6   salary_in_usd       83994 non-null  int64 
 7   employee_residence  83994 non-null  object
 8   remote_ratio        83994 non-null  int64 
 9   company_location    83994 non-null  object
 10  company_size        83994 non-null  object
dtypes: int64(4), object(7)
memory usage: 7.7+ MB

In [24]:
#df_cleaned.info()

unique_count = df_cleaned['job_title'].value_counts()
print(len(unique_count))

titles_above_one = unique_count[unique_count > 100]
print(titles_above_one)

6
job_title
Other                        37599
Data Scientist               12388
Data Engineer                10107
Software Engineer             9404
Data Analyst                  7866
Machine Learning Engineer     6226
Name: count, dtype: int64
